### ISSUE TITLE
JMX metrics from a container?

### ISSUE THREAD

**Issue by sschepens** [DataDog/docker-dd-agent#24]
Is it possible to get jmx metrics from a  containerized application?

**Comment by LotharSee** [comment]
HI @sschepens,

Yes that's possible. The main difficulty is to get the IP of your containerized application to put it in the Agent configuration. Depending on your cluster setup, they are multiple ways, the simpler being the usage of DNS.
Then you put the IP/port in the config the same way as with any host http://docs.datadoghq.com/integrations/java/

The Agent can run on the host or inside a container, both will work.

Is that helping you?

**Comment by sebest** [comment]
Hi,

Did anyone actually check that jmxfetch is working inside the container?

i don't see any support for java inside the container:
```
$ docker run -ti --entrypoint=bash datadog/docker-dd-agent
root@16aab141f1f3:/# find / | grep java
root@16aab141f1f3:/# find / | grep jdk
root@16aab141f1f3:/# find / | grep jre

On a side note the easier way to get the IP of another container is to use the "links" feature of docker, that way you can access the other container with its name.

**Comment by sschepens** [comment]
No, there is no support for jmxfetch in default image.
I created a Dockerfile extending from agent, installing java and enabling jmxfetch.
The thing with links is that they are unidirectional, we needed our application to link to datadog container to be able to post custom metrics, but we also needed jmxfetch to be able to connect to application to extract metrics.
We ended up with 3 containers: Application linked to Agent, JMXFetch linked to Application and Agent.
JMXFetch container is configured to only start jmxfetch and dogstatd, with dogstatd configured to forward every packet to Agent container.
It's not beautiful, but it works.

Hope it's useful.
Cheers!
